Functools partial reconstruct

voznesenskym opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117721

Adds a reconstruction for functools partial that cross graph breaks.

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

voznesenskym committed:
Functools partial reconstruct

fn stuff

fn stuff

[ghstack-poisoned]

voznesenskym committed:
Update on "Functools partial reconstruct"


fn stuff

fn stuff

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented on torch/_dynamo/variables/functions.py:
```diff
@@ -646,6 +646,21 @@ def get_val(v):
+        codegen(self.func)
+        if self.args and self.keywords:
+            # TODO(voz): Is there a util to help with this? neither the existing ones suffice?
+            unimplemented("Reconstruction of mixed args and keywords NYI")
```
@jansel do we need this? godbolt says python unpacks mixed kwargs/args as just args https://godbolt.org/z/5T6eKefrv but I don't know if that's always right...

jansel replied:
https://docs.python.org/3/library/dis.html#opcode-CALL_FUNCTION_EX

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented:
Updated - added test case for mix args and kwargs. Ty @jansel!

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented:
@pytorchbot rebase

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented:
@pytorchbot rebase

voznesenskym committed:
Update on "Functools partial reconstruct"


Adds a reconstruction for functools partial that cross graph breaks.

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

jansel commented:
Replaced by #118583
